### PR TITLE
feat(grammar): add variable definitions

### PR DESCRIPTION
**Pull request by Tirithel**

### CODE REVIEW

File: libs/tree-sitter-titan/grammar.js [hunk 1]
```diff
@@ -1,7 +1,42 @@
 module.exports = grammar({
-  name: 'titan',
-
+  name: "titan",
   rules: {
```
Comment (hasanaburayyan) [inline libs/tree-sitter-titan/grammar.js:3]: this way we can support multiple statements in a single file

```suggestion
  rules: {
    source_file: $ => repeat($._statement),

    _statement: ($) => choice($.variable_definition),
```

File: libs/tree-sitter-titan/grammar.js [hunk 2]
```diff
@@ -1,7 +1,42 @@
...
+    variable_declaration: ($) => seq(),
+
+    alpha_identifier: () =>
+      /[A-Za-z\$_][A-Z\$_a-z0-9]*(_[\-!#%&*+\/\\:<=>?@\u005e\u007c~]+)?/,
+
+    // TODO: Add annotations and modifiers and stuff
+    _variable_start: ($) => seq("let"),
+
+    // TODO: Add other types, maybe function/lambda
+    _type: ($) => choice($.literal_type),
```
Comment (hasanaburayyan) [inline libs/tree-sitter-titan/grammar.js]: Maybe we make this just a list of supported types until we add support for UserDefined types.

so in the meantime we dont allow 
```js
let x: Stringzs = 10
```
even though `Stringzs` could be a user defined class or interface etc... Atleast until we map out how we are going to handle user types.

So for no maybe just allow `string`, `num`, `bool`.

File: libs/tree-sitter-titan/grammar.js [hunk 3]
```diff
@@ -1,7 +1,42 @@
...
-  }
-});
\ No newline at end of file
+    variable_definition: ($) =>
+      seq(
+        $._variable_start,
+        field("pattern", $.alpha_identifier),
+        optional(seq(":", field("type", $._type))),
+        "=",
+        field("value", $._non_null_literal)
```
Comment (hasanaburayyan) [inline libs/tree-sitter-titan/grammar.js:13]: Should we end statements in semicolon?

Comment (Tirithel) [reply libs/tree-sitter-titan/grammar.js:13]: I'd like to make semicolon optional like how Scala does.

File: libs/tree-sitter-titan/grammar.js (hunk 2 above)
Comment (Tirithel) [reply libs/tree-sitter-titan/grammar.js]:
```suggestion
    _type: ($) => choice("string", "int", "bool"), // choice($.literal_type),
```

File: libs/tree-sitter-titan/grammar.js [hunk 4]
```diff
@@ -1,7 +1,45 @@
...
+    _variable_start: ($) => seq("let"),
+
+    // TODO: Add other types, maybe function/lambda
+    _type: ($) => choice("string", "int", "bool"), // choice($.literal_type),
+
+    // TODO: Add other types, string, float, etc
+    _non_null_literal: ($) => choice($.string, $.integer_literal),
+
+    // TODO: Replace with choice(simple string and multi line string)
+    string: ($) => $.alpha_identifier,
```
Comment (hasanaburayyan) [inline libs/tree-sitter-titan/grammar.js]:
```suggestion
    string: ($) => seq('"', $.alpha_identifier, '"'),
```

**Review by hasanaburayyan — APPROVED** [review]

**Pull request merged**